### DISCUSSION TITLE
Where is the location of Halide.h header file?

### DISCUSSION THREAD

**Discussion by paulrdelg** [halide/Halide#5578] [Q&A] [answered]
Apologies in advance if this question is very noob.

The tutorial here [lesson 01 basics](https://halide-lang.org/tutorials/tutorial_lesson_01_basics.html) says "The only Halide header file you need is Halide.h. It includes all of Halide."

When I search the directories (from the clone repo) for the file, I cannot seem to find it.

**Comment by abadams** [comment, marked as answer]
Halide.h is a build artifact, so if you build Halide you'll get one in the include directory of the install target.

**Reply by paulrdelg** [reply]
Also, following up your answer, when I build halide on one machine then create my app using halide library and compile that for another machine (for example, LLVM on my x86 machine and my app targeted for a RISCV machine), will this build artifact work properly?

**Reply by zvookin** [reply]
The intention is Halide on any host should be able to generate code for any of its targets and they should all work. (And given the exact same version of LLVM and Halide, should produce the same output on any host, though we do not test this is the case.)

RISC-V support is fairly new and I don't think it has been tested or used much. I expect we will need to fix at least a few things, but the intention is to make it a first class target. (The main issues is until the vector extension is finalized and reasonably ell supported by LLVM, we cannot take advantage of SIMD style parallelism.)
